### PR TITLE
fix(YieldFromArrayToYieldsFixer): skip touching empty array

### LINKED ISSUE #7270: `YieldFromArrayToYieldsFixer` improperly refactors yielding from an empty array
YieldFromArrayToYieldsFixer currently changes `yield from [];` to `yield;`. These are NOT equivalent: https://3v4l.org/RCLJa

### PR DESCRIPTION
**Pull request by keradus**
closes #7270

### CODE REVIEW

**Comment by keradus** [comment]
cc @enumag  , fyi @kubawerlos

**Comment by keradus** [comment]
fair question, let's add a test for it too

File: tests/Fixer/ArrayNotation/YieldFromArrayToYieldsFixerTest.php [hunk 1]
```diff
@@ -243,5 +243,48 @@ function foo() {
                 yield from [1, 2, 3];
             }',
         ];
+
+        yield 'from empty arrays' => [
```
Comment (Wirone) [inline tests/Fixer/ArrayNotation/YieldFromArrayToYieldsFixerTest.php:247]: I believe data set's name should represent the intent of the test case.

```suggestion
        yield 'skip empty arrays' => [
```

File: tests/Fixer/ArrayNotation/YieldFromArrayToYieldsFixerTest.php
Comment (Wirone) [inline tests/Fixer/ArrayNotation/YieldFromArrayToYieldsFixerTest.php:1]: Maybe it's an overkill, but what about:

```php
function foo1()
{
    yield from [
        // Inline comment,
        # and another one
    ];
}
```

- all comment styles covered
- multiline covered
- `,` is important in terms of this fixer (looking for items in the array), so maybe it's good to ensure that it's not matched within comment (even though it's rather impossible in terms of tokenizing 😝).

**Review by Wirone — CHANGES_REQUESTED** [review]

File: tests/Fixer/ArrayNotation/YieldFromArrayToYieldsFixerTest.php
Comment (keradus) [reply tests/Fixer/ArrayNotation/YieldFromArrayToYieldsFixerTest.php:1]: if you consider this test missing to be a blocker for PR, happy to add it

yeah, not sure about `,` in context of tokenizer, but won't hurt so I'm ok to add it

**Review by Wirone — APPROVED** [review]

**Pull request merged**